docs/enhanced-provider-k8s.md

```diff
@@ -0,0 +1,314 @@
+
+However, as the `ClusterServiceVersion` resource still exists, so does the `InstallPlan` resource, the Kong operator pod still remains. This is not considered as a clean uninstall since some garbage resources remain in the cluster. For more details on why it is designed in such a way, you can refer to [this issue](https://github.com/operator-framework/operator-lifecycle-manager/issues/1168) from OLM community.
+
+![](images/unmanaged-resource.png)
+
+We call the external resource such as `ClusterServiceVersion` and `InstallPlan` in this case are **unmanaged external resources** which means these resources should have been managed by the provider, so that when the managed resource is deleted, the corresponding external resource will be deleted accordingly.
+
+Apparently, the solution to this problem is turning unmanaged external resource into managed one. In our case, this can be done by defining an `Object` resource to represent the `ClusterServiceVersion` resource. Then, the deletion of `Object` resource will trigger the deletion of `ClusterServiceVersion` resource. There is no need to define `Object` for `InstallPlan` because when `ClusterServiceVersion` is deleted, it will trigger the deletion of `InstallPlan` automatically which is handled by OLM.
+
+One interesting thing for the `Object` resource to be defined is that, the provider only needs to take care of the deletion and observation for the `ClusterServiceVersion` resource. It should not be responsible for the creation and update because these operations are handled by OLM. This scenario leads to the introduction of the feature **Resource Management Policy** in Provider Kubernetes.
+
+### Resource Management Policy
+
+The resource management policy is used to instruct the provider to manage Kubernetes resources in a fine-grained way. Given that:
+
+* To manage an external resource that can not be observed is usually meaningless.
+* Create and update are usually coupled together.
+
+We only need 4 types of management policies:
+
+| Observe | Create & Update | Delete | Management Policy   | Memo
+|:--------|:----------------|:-------|:--------------------|:----
+| Y       | Y               | Y      | Default             | The provider can fully manage the resource. This is the default policy.
+| Y       | Y               | N      | ObserveCreateUpdate | The provider can observe, create, or update the resource, but can not delete it. 
+| Y       | N               | Y      | ObserveDelete       | The provider can observe or delete the resource, but can not create and update it.
+| Y       | N               | N      | Observe             | The provider can only observe the resource. This maps to the read-only scenario where the resource is fully controlled by third party application and the provider only reads the status of the resource.
```

Comment:
> and the provider only reads the status of the resource.

Indeed, the provider reads the whole resource manifest (and stores at `Object` status.atProvider) not just status right?

Reply:
My fault, fixed.